sub-packages/bionemo-core/src/bionemo/core/data/load.py

```diff
@@ -203,14 +200,34 @@ def load(
         downloader=download_fn,
         processor=processor,
     )
+    return Path(download)
 
-    # Pooch by default returns a list of unpacked files if they unpack a zipped or tarred directory. Instead of that, we
-    # just want the unpacked, parent folder.
-    if isinstance(download, list):
-        return Path(processor.extract_dir)  # type: ignore
 
-    else:
-        return Path(download)
+class CachedProcessor:
+    def __call__(self, fname, action, pooch):
+        if self.extract_dir is None:
+            self.extract_dir = fname + self.suffix
+        else:
+            archive_dir = fname.rsplit(os.path.sep, maxsplit=1)[0]
+            self.extract_dir = os.path.join(archive_dir, self.extract_dir)
+
+        from pathlib import Path
+
+        if not Path(self.extract_dir).exists():
+            super().__call__(fname, action, pooch)
```

Comment:
Didn't we already make a change here where we check to see if `Path(str(self.extract_dir)+".lock")` exists and make it and all that? Double check that is pushed up.

From memory the logic was like:
```
lockfile = Path(str(self.extract_dir)+".lock")
if not Path(self.extract_dir).exists() or lockfile.exists():
    lockfile.touch()
    super().__call__(fname, action, pooch)
    lockfile.unlink() # remove the lockfile if the previous command was successful, otherwise re-try next time.